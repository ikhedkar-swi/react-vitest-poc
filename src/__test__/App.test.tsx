import { expect, test } from "vitest";
import { create } from "react-test-renderer";
import App from "../App";

test("renders learn react link", async () => {
  const app = create(<App />);
  const instance = app.root;
  const learnReactLink = await instance.findByProps({ className: "App-link" });

  expect(learnReactLink.props).toEqual({
    children: "Learn React",
    className: "App-link",
    href: "https://reactjs.org",
    rel: "noopener noreferrer",
    target: "_blank",
  });
  expect(app).toMatchSnapshot();
});
