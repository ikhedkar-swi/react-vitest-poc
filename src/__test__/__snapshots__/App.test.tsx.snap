// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders learn react link 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="/src/logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;
